sklearn/model_selection/_search.py

```diff
@@ -538,10 +560,10 @@ def decision_function(self, X):
             underlying estimator.
 
         """
-        self._check_is_fitted("decision_function")
+        check_is_fitted(self)
```

Comment:
we should either use `_check_refit` in all of these functions or none, don't we?

Reply:
`_estimator_has` will call `_check_refit`

Reply:
This is the beauty of the decorator, the inner code look-alike any other estimator because the check is done by the decorator. So only the fitting should be checked here.